File: IDeal.cs
using System;

public class IDeal
{
    private string iban;
    private string bankName;
    private string accountHolderName;

    public string GetIban()
    {
        return iban;
    }

    public void SetIban(string iban)
    {
        this.iban = iban;
    }

    public string GetBankName()
    {
        return bankName;
    }

    public void SetBankName(string bankName)
    {
        this.bankName = bankName;
    }

    public string GetAccountHolderName()
    {
        return accountHolderName;
    }

    public void SetAccountHolderName(string accountHolderName)
    {
        this.accountHolderName = accountHolderName;
    }

    public bool IsValidIban()
    {
        if (iban.Length < 5 && iban.Length > 32)
        {
            return false;
        }
        if (!iban.StartsWith("^[A-Z]{2}[0-9]{2}$"))
        {
            return false;
        }
        if (!Regex.IsMatch(iban.Substring(4), "^[A-Z0-9]+$"))
        {
            return false;
        }
        return true;
    }

    public bool IsValidBank()
    {
        if (!Regex.IsMatch(bankName, "^[a-zA-Z]+$"))
        {
            return false;
        }
        return true;
    }

    public bool IsValidAccountName()
    {
        if (!Regex.IsMatch(accountHolderName, "^[a-zA-Z]+$"))
        {
            return false;
        }
        return true;
    }
}

File: IDeal.java
public class IDeal {
    private String iban;
    private String bankName;
    private String accountHolderName;

    public String getIban() {
        return iban;
    }

    public void setIban(String iban) {
        this.iban = iban;
    }

    public String getBankName() {
        return bankName;
    }

    public void setBankName(String bankName) {
        this.bankName = bankName;
    }

    public String getAccountHolderName() {
        return accountHolderName;
    }

    public void setAccountHolderName(String accountHolderName) {
        this.accountHolderName = accountHolderName;
    }

    public boolean isValidIban(){
        if (iban.length() < 5 && iban.length() > 32){
            return false;
        }
        if (!iban.matches("^[A-Z]{2}[0-9]{2}")) {
            return false;
        }
        if (!iban.substring(4).matches("^[A-Z0-9]+$")){
            return false;
        }
        return true;
    }
    public boolean isValidBank() {
        if (!bankName.matches("^[a-zA-Z]+$")){
            return false;
        }
        return true;
    }

    public boolean isValidAccountName() {
        if (!accountHolderName.matches("^[a-zA-Z]+$")){
            return false;
        }
        return true;
    }
}
